### PR TITLE
Switches project infrastructure from Maven to Gradle

### PR DESCRIPTION
**Pull request by cschierle**
Fixes: [SIRI-635](https://scireum.myjetbrains.com/youtrack/issue/SIRI-635)

### Breaking changes:
- Maven setup (e.g. `pom.xml`) has been replaced by Gradle equivalent
- Drone config in `drone.yml` has been updated to use Gradle tasks for validation and publication
- Removed dependency on `parsii` and replaced its usages with corresponding references from sirius-kernel

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    api("org.elasticsearch.client:elasticsearch-rest-client:8.4.1")
+    // Required as the version brought by elasticsearch-rest-client contains security issues
+    api("org.apache.httpcomponents:httpclient:4.5.13")
+
+    testImplementation("org.mariadb.jdbc:mariadb-java-client:3.0.6")
+
+    testImplementation("com.clickhouse:clickhouse-jdbc:0.3.2-patch11")
+    // Required as the version brought by clickhouse-jdbc contains security issues
+    testImplementation("com.fasterxml.jackson.core:jackson-databind")
```
Comment (mkeckmkeck) [inline build.gradle.kts:41]: This required make make gradle use the version from bom, and not the explicitly defined, transitive version from clickhouse?

File: build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+plugins {
+    id("java-library")
+    id("com.scireum.sirius-parent") version "11.0.5"
+    id("org.sonarqube") version "3.4.0.2513"
+}
+
+apply(plugin = "com.scireum.sirius-parent")
+
+dependencies {
+    api("com.scireum:sirius-kernel:${property("sirius-kernel")}")
```
Comment (mkeckmkeck) [inline build.gradle.kts:18]: Is this kind of a best practice naming for gradle properties? I would prefer a more descriptive name containing version, like version-sirius-kernel

File: src/test/java/TestSuite.java [hunk 3]
```diff
@@ -7,11 +7,28 @@
...
 import org.junit.runner.RunWith;
 import sirius.kernel.ScenarioSuite;
+import sirius.kernel.TestHelper;
 
 @RunWith(ScenarioSuite.class)
-@SuiteClasses({"**/*Test.class", "**/*Spec.class"})
+@SuiteClasses("**/*Spec.class")
 public class TestSuite {
 
+    @BeforeClass
```
Comment (mkeckmkeck) [inline src/test/java/TestSuite.java:25]: Why is this change required? Doesn't this happen in the ScenarioSuite already?

**Review by idlira — APPROVED** [review]
